src/test/scala/com/fulcrumgenomics/bam/ClipBamTest.scala

```diff
@@ -166,21 +166,6 @@ class ClipBamTest extends UnitSpec with ErrorLogLevel with OptionValues {
     r1.end >= r2.start shouldBe false
   }
 
-  // This is a weird test that ensures that things terminate correctly when due to highly clipped
-  // input reads, one of the reads becomes unmapped during clipping
-  it should "handle reads that get unmapped because they are fully overlapped" in {
-    val builder = new SamBuilder(readLength=50)
-    val clipper = new ClipBam(input=dummyBam, output=dummyBam, ref=ref, clipOverlappingReads=true)
-    val Seq(r1a, r2a) = builder.addPair(name="q1", start1=100, start2=100, cigar1="3M47S", cigar2="48S2M")
-    val Seq(r1b, r2b) = builder.addPair(name="q2", start1=100, start2=100, cigar1="2M48S", cigar2="47S3M")
-
-    for ((r1, r2) <- Seq((r1a, r2a), (r1b, r2b))) {
-      clipper.clipPair(r1, r2)
-      val ok = r1.unmapped || r2.unmapped || r1.end < r2.start
-      ok shouldBe true
-    }
-  }
```

Comment:
where did this test case get moved to?

Reply:
This specific test may no longer be necessary since trimming shouldn't fully trim a read anymore.  Specifically, the result now evaluates to `false` because there shouldn't be a way for a read to be entirely trimmed.  That's actually how I tracked down the second regression (found an entirely trimmed read).

Reply:
This makes me a little nervous.  I get this test no longer causes a read to be unampped, but is it now theoretically impossible to fully clip a read?  What if the read is e.g. post indel realignment and has a CIGAR that is mostly/all insertion with 1bp overlapping a mate?

Reply:
A different test could be written (not sure if this is the best place for it or not), but clipping based on `clipOverlappingReads=true` shouldn't ever fully clip a read, unless maybe you have two 1bp reads that are overlapped?

Reply:
Fair enough.